app/ts/tests/test.ts

```diff
@@ -0,0 +1,402 @@
+		
+		if (priorContentHash === afterContentHash) throw new Error('content has was not set')
+		if (decodeContentHash(afterContentHash) !== testContentHash) throw new Error('wrong content hash')
+		
+		const priorAddr = dataStringWith0xStart(result[0].calls[1].returnData)
+		const afterAddr = dataStringWith0xStart(result[0].calls[4].returnData)
+		if (BigInt(priorAddr) === BigInt(afterAddr)) throw new Error('addr was not set')
+		if (BigInt(afterAddr) !== ownAddress) throw new Error('addr is not correct')
+	}
+	const testThatFusesAreCorrect3PartAgain = async () => {
+		const ownedTokens3 = [BigInt(namehash(subdomainRouteNames3[0]))]
+		const ownedTokens3_another = [BigInt(namehash(subdomainRouteNames3[0])), BigInt(namehash(subdomainRouteNames3[1]))]
+		//check that we can create more subdomains after creating noe
+		const result = await ethSimulateTransactions(rpc, [
+			makeImmutableDomain(subdomainRouteNames3, ownedTokens3),
+			makeImmutableDomain(subdomainRouteNames3_another, ownedTokens3_another),
+			{
+				from: whale,
+				to: BigInt(ENS_TOKEN_WRAPPER),
+				input: stringToUint8Array(encodeFunctionData({
+					abi: ENS_WRAPPER_ABI,
+					functionName: 'getData',
+					args: [BigInt(namehash(subdomainRouteNames3_another[2]))]
+				})),
+			},
+			{
+				from: ownAddress,
+				to: BigInt(ENS_TOKEN_WRAPPER),
+				input: stringToUint8Array(encodeFunctionData({
+					abi: ENS_WRAPPER_ABI,
+					functionName: 'balanceOf',
+					args: [addressString(BURN_ADDRESS), BigInt(namehash(subdomainRouteNames3_another[2]))]
+				}))
+			},
+		])
+		if (!allSuccess(result)) throw new Error('transaction failed')
+		if (result[0]?.calls[2]?.returnData === undefined || result[0]?.calls[3]?.returnData === undefined) throw new Error('no call results')
+		// child
+		const childData = decodeFunctionResult({ abi: ENS_WRAPPER_ABI, functionName: 'getData', data: dataStringWith0xStart(result[0]?.calls[2]?.returnData) })
+		const childFuses = new Set(extractENSFuses(BigInt(childData[1])))
+		if (!areSetsEqual(childFuses, new Set(FINAL_CHILD_FUSES))) throw new Error('not correct child fuses')
+		if (bytesToUnsigned(result[0]?.calls[3]?.returnData) !== 1n) throw new Error('burn address does not have the child')
+	}
+	const testAnyoneCanRenew = async() => {
+		throw new Error('TODO, RENEWALS DO NOT WORK')
+		/*,
+		{
+			from: whale,
+			to: BigInt(ENS_TOKEN_WRAPPER),
+			input: stringToUint8Array(encodeFunctionData({
+				abi: ENS_WRAPPER_ABI,
+				functionName: 'extendExpiry',
+				args: [namehash('darkflorist.eth'), labelhash('immutable'), 2n ** 64n - 1n]
+			})),
+		}*/
+	}
+	await testMakeImmutable()
+	await testWeHaveTheOrignalToken()
+	await testThatFusesAreCorrect()
+	await testNoData()
+	await testNoNonBatchTransfer()
+	await testThatFusesAreCorrect3Part()
+	await testContentHashIsSet()
+	await testThatFusesAreCorrect3PartAgain()
+	await testAnyoneCanRenew()
```

Comment:
You could run these all in parallel I think.  Since they are done with `eth_simulate`, each one will execute in a separate context.

Reply:
That generally makes debugging more difficult as many things can go wrong at once. Running these tests is quite fast anyway so I don't think it makes sense to optimize them